returnn/datasets/distrib_files.py

```diff
@@ -283,13 +287,15 @@ def init_seq_order(self, epoch: Optional[int] = None, seq_list=None, seq_order=N
             else:
                 raise ValueError(f"{self}: seq_ordering {self.seq_ordering!r} not supported")
             file_bins = self._distribute_evenly_by_size(
-                num_bins=self._num_shards * self.partition_epoch,
+                num_bins=self._num_data_shards * self._num_file_shards * self.partition_epoch,
                 file_sizes=self._file_sizes,
                 files_order=files_order_flat,
             )
-            self_index_base = self.partition_epoch * self._shard_index
-            self_index_end = self_index_base + self.partition_epoch
-            self._files_order_cache[full_epoch_0idx_] = file_bins[self_index_base:self_index_end]
+            self_index_base = self._num_data_shards * self.partition_epoch * self._file_shard_index
+            self_index_end = self_index_base + (self.partition_epoch * self._num_data_shards)
+            self._files_order_cache[full_epoch_0idx_] = file_bins[self_index_base:self_index_end][
+                self._data_shard_index :: self._num_data_shards
```

Comment:
Here do again do this inconsistently / more complex than it needs to be. Same case as in `_apply_partition_epoch`, we can just reuse the existing logic.